base_classes/NXparameters.nxdl.xml

```diff
@@ -26,11 +26,69 @@
     xsi:schemaLocation="http://definition.nexusformat.org/nxdl/3.1 ../nxdl.xsd"
     name="NXparameters" 
 	type="group" extends="NXobject">
-    <doc>Container for parameters, usually used in processing or analysis.</doc>
+    <doc>
+        Container for parameters used in processing or analysing data.
+
+        This base class can be used in a number of different contexts,
+        and is allowed in any other base class. However, a common use
+        case is to store the results of optimizing parameters, for
+        example, by least-squares fitting, so a number of attributes
+        have been defined for this purpose. Normally, this group will be
+        stored in a :ref:NXprocess group, which defines the program used
+        to process these parameters. If multiple functions are used to
+        analyze the data, there can be additional NXparameter groups for
+        each one.        
+    </doc>
+    <attribute name="model">
```

Comment:
Personally would make this a field as then you can add version of the model for an attribute as attribute of an attribute is not possible in NeXus, in FAIRmat sometimes we use e.g. as a field "algorithm" same idea like model.

Reply:
Exactly these were also my thoughts

Reply:
Maybe then we should mention algorithm as well in the docstring and then that is fine